sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -690,6 +697,9 @@ class TungstenAggregationIterator(
         val mapMemory = hashMap.getPeakMemoryUsedBytes
         val sorterMemory = Option(externalSorter).map(_.getPeakMemoryUsedBytes).getOrElse(0L)
         val peakMemory = Math.max(mapMemory, sorterMemory)
+        totalPeakMemory += peakMemory
+        maxPeakMemory += peakMemory
+        totalSpilledSize += TaskContext.get().taskMetrics().memoryBytesSpilled - spilledSizeBefore
```

Comment:
@andrewor14 is it any place that needs to create multiple threads in a task? Currently, accumulators are not thread-safe. I assume there is only one thread updating accumulators and one thread collecting accumulators' values when running a task. If there are multiple updating threads, we need to make accumulators thread-safe.